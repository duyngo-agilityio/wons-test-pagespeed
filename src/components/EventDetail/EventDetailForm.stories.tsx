import type { Meta, StoryObj } from '@storybook/react';

// components
import EventDetail from './index';

const meta: Meta<typeof EventDetail> = {
  title: 'Components/EventDetail',
  component: EventDetail,
  parameters: {
    layout: 'centered',
    controls: { expanded: true },
  },
  args: {
    title: 'Event Title',
    time: '10:00 AM',
    location: '123 Main St',
    guests: [
      { name: 'John Doe', avatar: '/avatars/john.png' },
      { name: 'Jane Doe', avatar: '/avatars/jane.png' },
    ],
  },
};

export default meta;
type Story = StoryObj<typeof EventDetail>;

export const Default: Story = {};
