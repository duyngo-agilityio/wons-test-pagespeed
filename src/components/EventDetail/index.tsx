'use client';
import { memo } from 'react';
import isEqual from 'react-fast-compare';

// components
import {
  Button,
  ImageFallback,
  Text,
  Heading,
  FaLocationDot,
  ImPhone,
} from '@/components';

// icons
import { FaUsers } from 'react-icons/fa';

// Constants
import { IMAGES } from '@/constants';

export interface Guest {
  name: string;
  avatar: string;
}

export interface EventCardProps {
  title: string;
  time: string;
  location: string;
  guests: Guest[];
  link: string;
}

const EventDetail = ({
  title,
  time,
  location,
  guests,
  link,
}: EventCardProps) => {
  const handleJoinGoogleMeet = () => {
    // TODO: will handle again later
    window.open('https://meet.google.com/', '_blank');
  };
  return (
    <div className="max-w-md mx-auto bg-white dark:bg-gray-400 rounded-lg overflow-hidden">
      <div className="p-4">
        <div className="mb-8">
          <Heading title={title} className="mb-2" />
          <Text size="2xl" className="opacity-70 font-medium" text={time} />
        </div>

        <div className="flex items-center mt-4">
          <FaLocationDot
            className="text-blue-800/30 dark:text-white/30"
            fontSize={16}
          />
          <Text
            as="span"
            text={location}
            size="xl"
            className="opacity-70 font-medium ml-4"
          />
        </div>

        <div className="flex items-center mt-4">
          <ImPhone
            className="text-blue-800/30 dark:text-white/30"
            fontSize={16}
          />
          <Text
            size="xl"
            className="opacity-70 font-medium ml-4"
            text="Join by phone"
          />
        </div>

        <div className="flex flex-col items-start mt-4">
          <div className="flex items-center">
            <ImageFallback
              src={IMAGES.GOOGLE_MEET}
              alt="google meet"
              width={16}
              height={16}
            />
            <Button
              onClick={handleJoinGoogleMeet}
              className="!bg-blue-500 dark:!bg-blue-500 text-white font-bold py-2 px-4 rounded ml-4"
            >
              <Text
                size="xl"
                className="text-white dark:text-dark font-medium"
                text="Join with Google Meet"
              />
            </Button>
          </div>
          <Text
            size="xl"
            className="opacity-70 font-medium mt-1 text-2xl ml-[33px]"
            text={link}
          />
        </div>

        <div className="flex items-center mt-4">
          <FaUsers
            className="text-blue-800/30 dark:text-white/30"
            fontSize={18}
          />
          <Text
            size="xl"
            className="opacity-70 font-medium ml-4"
            text={`${guests.length} guest${guests.length !== 1 ? 's' : ''}`}
          />
        </div>
      </div>

      <div className="px-8 ml-4">
        <ul className="list-none">
          {guests.map((guest, index) => (
            <li
              key={index}
              className="flex items-center mb-2 text-gray-600 dark:text-gray-300"
            >
              <ImageFallback
                src={guest.avatar}
                width={30}
                height={30}
                className="rounded-full mr-2"
                alt={`${guest.name}'s avatar`}
              />
              <Text
                size="xl"
                className="opacity-70 font-medium"
                text={guest.name}
              />
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default memo(EventDetail, isEqual) as <T>(
  props: EventCardProps & T,
) => JSX.Element;
