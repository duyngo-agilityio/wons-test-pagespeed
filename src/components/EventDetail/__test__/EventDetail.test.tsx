import EventDetail from '../index';

describe('EventDetail Component', () => {
  it('should match snapshot', () => {
    const { container } = testLibJestUtils.render(
      <EventDetail
        title="Upcoming Event"
        time="10:00 AM"
        location="123 Main St"
        link="https://meet.google.com/"
        guests={[
          { name: 'John Doe', avatar: '/avatars/john.png' },
          { name: 'Jane Doe', avatar: '/avatars/jane.png' },
        ]}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
