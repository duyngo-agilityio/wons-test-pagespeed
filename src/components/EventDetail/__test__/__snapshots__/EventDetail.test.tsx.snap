// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetail Component should match snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
    <div
      class="max-w-md mx-auto bg-white dark:bg-gray-400 rounded-lg overflow-hidden"
    >
      <div
        class="p-4"
      >
        <div
          class="mb-8"
        >
          <h2
            aria-level="2"
            class="font-bold font-dm-sans text-blue-800 dark:text-white text-5xl mb-2"
          >
            Upcoming Event
          </h2>
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-2xl opacity-70 font-medium"
          >
            10:00 AM
          </p>
        </div>
        <div
          class="flex items-center mt-4"
        >
          <svg
            class="text-blue-800/30 dark:text-white/30"
            fill="currentColor"
            font-size="16"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M215.7 499.2C267 435 384 279.4 384 192C384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2c12.3 15.3 35.1 15.3 47.4 0zM192 128a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"
            />
          </svg>
          <span
            class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium ml-4"
          >
            123 Main St
          </span>
        </div>
        <div
          class="flex items-center mt-4"
        >
          <svg
            class="text-blue-800/30 dark:text-white/30"
            fill="currentColor"
            font-size="16"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 10c-1 1-1 2-2 2s-2-1-3-2-2-2-2-3 1-1 2-2-2-4-3-4-3 3-3 3c0 2 2.055 6.055 4 8s6 4 8 4c0 0 3-2 3-3s-3-4-4-3z"
            />
          </svg>
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium ml-4"
          >
            Join by phone
          </p>
        </div>
        <div
          class="flex flex-col items-start mt-4"
        >
          <div
            class="flex items-center"
          >
            <img
              alt="google meet"
              data-nimg="1"
              decoding="async"
              height="16"
              loading="lazy"
              src="http://localhost/_next/image?url=%2Fgoogle-meet.webp&w=32&q=75"
              srcset="/_next/image?url=%2Fgoogle-meet.webp&w=16&q=75 1x, /_next/image?url=%2Fgoogle-meet.webp&w=32&q=75 2x"
              style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="16"
            />
            <button
              class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default data-[hover=true]:opacity-hover border-transparent dark:text-white hover:!bg-gray-200/50 dark:hover:!bg-gray-900 text-md border-[1px] h-auto min-w-max !bg-blue-500 dark:!bg-blue-500 text-white font-bold py-2 px-4 rounded ml-4"
              type="button"
            >
              <div
                class="flex items-center"
              >
                <p
                  class="font-dm-sans text-blue-800 dark:text-white text-xl text-white dark:text-dark font-medium"
                >
                  Join with Google Meet
                </p>
              </div>
            </button>
          </div>
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium mt-1 text-2xl ml-[33px]"
          >
            https://meet.google.com/
          </p>
        </div>
        <div
          class="flex items-center mt-4"
        >
          <svg
            class="text-blue-800/30 dark:text-white/30"
            fill="currentColor"
            font-size="18"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 640 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
            />
          </svg>
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium ml-4"
          >
            2 guests
          </p>
        </div>
      </div>
      <div
        class="px-8 ml-4"
      >
        <ul
          class="list-none"
        >
          <li
            class="flex items-center mb-2 text-gray-600 dark:text-gray-300"
          >
            <img
              alt="John Doe's avatar"
              class="rounded-full mr-2"
              data-nimg="1"
              decoding="async"
              height="30"
              loading="lazy"
              src="http://localhost/_next/image?url=%2Favatars%2Fjohn.png&w=64&q=75"
              srcset="/_next/image?url=%2Favatars%2Fjohn.png&w=32&q=75 1x, /_next/image?url=%2Favatars%2Fjohn.png&w=64&q=75 2x"
              style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="30"
            />
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium"
            >
              John Doe
            </p>
          </li>
          <li
            class="flex items-center mb-2 text-gray-600 dark:text-gray-300"
          >
            <img
              alt="Jane Doe's avatar"
              class="rounded-full mr-2"
              data-nimg="1"
              decoding="async"
              height="30"
              loading="lazy"
              src="http://localhost/_next/image?url=%2Favatars%2Fjane.png&w=64&q=75"
              srcset="/_next/image?url=%2Favatars%2Fjane.png&w=32&q=75 1x, /_next/image?url=%2Favatars%2Fjane.png&w=64&q=75 2x"
              style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="30"
            />
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium"
            >
              Jane Doe
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
